src/Views/KeyboardLayoutView.vala

```diff
@@ -78,22 +78,34 @@ public class KeyboardLayoutView : AbstractInstallerView {
 
         back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);
 
-        next_button.clicked.connect (() => next_step ());
+        next_button.clicked.connect (() => {
+            unowned Gtk.ListBoxRow row = input_variant_widget.main_listbox.get_selected_row ();
+            if (row != null) {
+                var layout = ((LayoutRow) row).layout;
+                Configuration.get_default ().keyboard_layout = layout.name;
+
+                unowned Gtk.ListBoxRow vrow = input_variant_widget.variant_listbox.get_selected_row ();
+                if (vrow != null) {
+                    string variant = ((VariantRow) vrow).code;
+                    Configuration.get_default ().keyboard_variant = variant;
+                } else if (layout.variants.is_empty) {
+                    Configuration.get_default ().keyboard_variant = null;
+                } else {
+                    row.activate();
+                    return;
+                }
+            } else {
+                error ("next_button enabled when no keyboard selected");
```

Comment:
You should not use `error ()` in GUI applications, as this function will just abort the process and make the window close, even if it's really unlikely that this if branch gets executed.